### PR TITLE
Add support for dedup, dedup_by, and dedup_by_key

### PR DESCRIPTION
**Pull request by hoggetaylor**
This change is [Reviewable](https://reviewable.io/reviews/servo/rust-smallvec/72)

### CODE REVIEW

**Comment by hoggetaylor** [comment]
This pull request adds support for dedup, dedup_by, and dedup_by_key as requested in #2

File: lib.rs [hunk 1]
```diff
@@ -91,13 +91,37 @@ pub trait VecLike<T>:
 
     /// Append an element to the vector.
     fn push(&mut self, value: T);
+    /// Removes consecutive repeated elements.
+    fn dedup(&mut self) where T: PartialEq<T>;
```
Comment (mbrubeck) [inline lib.rs]: I'm not sure we want to add these as trait methods.  Currently this trait is very minimal; adding these methods would be a breaking change and a bit of a burden for other types that implement it.  (The use case for this trait is questionable anyway, and I think we might deprecate and remove it instead.)

If we did add these methods, then we should probably also add `truncate`, and then provide default implementations of the `dedup` methods based on that.

Comment (hoggetaylor) [reply lib.rs]: I have added a commit to remove these methods from the VecLike trait.

**Comment by mbrubeck** [comment]
@bors-servo r+

Thanks!

**Pull request merged**